engine/src/input/axis.cpp

```diff
@@ -60,11 +61,29 @@ float InputAxis::value() const
 
 void InputAxis::value(float value)
 {
-    mValue = value;
+    mValue = std::abs(value) < mDeadzone ? 0.0F : value;
 
     if (std::abs(mValue) > 1.0F)
     {
         CUBOS_WARN("Axis value out of range: {}", mValue);
         mValue = mValue > 1.0F ? 1.0F : -1.0F;
     }
 }
+
+float InputAxis::deadzone() const
+{
+    return mDeadzone;
+}
+
+void InputAxis::deadzone(float deadzone)
+{
+    if (deadzone < 0.0F || deadzone > 1.0F)
+    {
+        CUBOS_WARN("Invalid deadzone value: {}. Value must be between 0.0 and 1.0.", deadzone);
+        mDeadzone = deadzone > 1.0F ? 1.0F : 0.0F;
+    }
+    else
+    {
+        mDeadzone = deadzone;
+    }
```

Comment:
```suggestion
if (deadzone < 0.0F || deadzone > 1.0F)
{
    CUBOS_WARN("Invalid deadzone value: {}. Value must be between 0.0 and 1.0.", deadzone);
}
mDeadzone = std::clamp(deadzone, 0.0F, 1.0F);
```

Just a small nitpick but this may look cleaner and make variable only be changed in one line of the function